### ISSUE TITLE
2d image mask support

### ISSUE THREAD

**Issue by briancantwe** [pierotofy/OpenSplat#35]
Hello, and congratulations on your awesome work!  Now that you've added Colmap support, I'm really excited to dive in.  

Question, do you plan to support 2d image masks?   These would be used to ignore specific regions of the image during training. 

Thanks so much!

**Comment by pierotofy** [comment]
Hey, :wave: this would be a cool feature for sure; while my focus is going to be on other priorities in the short term, we'd welcome a PR that adds support for image masks!

**Comment by NotCompsky** [comment]
I don't know anything about gaussian splatting, but if it's not too complicated I might be able to implement it.

What would it involve?

If I were to (A) simply turn pixels black for unwanted areas, I assume that wouldn't work well, because the result would be lots of unwanted black gaussians, correct?

If I were to (B) turn each pixel into a random colour, and change the colour randomly each step, I assume this would result in the training discarding those regions, correct?

(B) can be implemented without knowing anything about the maths of gaussian splatting (so I could do it), but it's probably far less efficient than the proper way of doing it.

**Comment by pierotofy** [comment]
At its core I think it's a matter of zeroing out the values of both rendered and ground truth images before computing the error (rendered - gt).

See how splatfacto does it. https://github.com/nerfstudio-project/nerfstudio/blob/main/nerfstudio/models/splatfacto.py#L665

There might be other things to consider though, I haven't looked deeply into it.

**Comment by NotCompsky** [comment]
Thanks!

Zeroing rendered and gt seems to work perfectly for my masked dataset.

The results are not a higher visual quality than the unmasked splats were - but it removes unwanted splats, and it trains much faster.

The error (i.e. loss) is very low (`mainLoss.item` often below 0.0006) - because most of each `rendered` and `gt` is zeroed out. Do you think the error might get so low that it causes precision errors (or floating point errors or something of that kind) that make it difficult to find a minima during backpropagation?

**Comment by pierotofy** [comment]
It makes sense that the loss would be lower; I guess it depends on the size of the masked area.

I'm unsure about the effects that such a small loss would have on precision, but you could try to modify the PSNR/SSIM functions to ignore the mask areas https://github.com/pierotofy/OpenSplat/blob/main/model.cpp#L45-L52 and see if you get better results.

**Comment by NotCompsky** [comment]
I think it would be fine if we just by crop the image to the mask's bounding box.

Rare edge cases (e.g. a splat of lots of small objects) might have this problem, but if someone
